src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -1456,35 +1479,17 @@ class CaseExpressionSegment(BaseSegment):
         Sequence(
             "CASE",
             Indent,
-            AnyNumberOf(
-                Sequence(
-                    "WHEN",
-                    Indent,
-                    Ref("ExpressionSegment"),
-                    "THEN",
-                    Ref("ExpressionSegment"),
-                    Dedent,
-                )
-            ),
-            Sequence("ELSE", Indent, Ref("ExpressionSegment"), Dedent, optional=True),
+            AnyNumberOf(Ref("WhenClauseSegment")),
+            Ref("ElseClauseSegment", optional=True),
             Dedent,
             "END",
         ),
         Sequence(
             "CASE",
             OneOf(Ref("ExpressionSegment")),
```

Comment:
Oh yes, that looks like a misunderstanding by the original author. I'll remove it and confirm nothing breaks.

Reply:
I think you only removed the OneOf but could we merge the two different case statements together as part of this? Don’t see the need to have two separate repetitions with just ones expression different between them - just make that expression optional.

And, I appreciate this was not added by this PR but may as well clean up as part of this if it’s simple. If it’s not as simple as I’m implying then happy to leave as was.

Reply:
I tried changing it to this, but I'm getting parse errors on some of the test cases. I suspect this is one of those things where the core parser isn't quite working right, and perhaps adding some hints might fix it. Seems like it's more involved than we hoped, so I would prefer to defer this.

Note that this PR is already a "side journey" of a larger PR, #2126, and some related early work on adding new `CASE`-related rule, #1799. Simplifying rule creation will be 💯. The PR is also ready for a first look when you have time. I'd like to merge this one first, because these dialect changes will make all 3 `CASE`-related rules (2 existing ones plus a new one) more amenable to being simplified. 💪